### ISSUE TITLE
[Feature Request] To change the way to deal with the logarithm of standard deviation for SAC

### ISSUE THREAD

**Issue by lzhyu** [DLR-RM/stable-baselines3#620]
### 🚀 Feature

 To change the way to deal with the logarithm of standard deviation for SAC.

### Motivation

in  [sac/policies.py](https://github.com/DLR-RM/stable-baselines3/blob/master/stable_baselines3/sac/policies.py) line 169-171, we use
```python 
        log_std = self.log_std(latent_pi)
        # Original Implementation to cap the standard deviation
        log_std = th.clamp(log_std, LOG_STD_MIN, LOG_STD_MAX)
```
However, this may lead to zero gradients when log_std is out of range due to `torch.clamp`. 

### Alternatives

Replace code above with
```python
        log_std = torch.tanh(log_std)
        log_std = LOG_STD_MIN + 0.5 * (
            LOG_STD_MAX  - LOG_STD_MIN 
        ) * (log_std + 1)
``` 
as in [rad](https://github.com/MishaLaskin/rad/blob/master/curl_sac.py) line 81-84.

**Comment by Miffyli** [comment]
Thanks for the suggestion! Is there formal comparison somewhere between these two? Intuitively, the latter could be better, but more often than not the intuition does not match with the results :). Since this would be a silent change that could affect training results, I am not 100% about adding this (silently "breaks" backwards compatibility, does not match original SAC implementation necessirely).

@araffin ?
